lib/ansible/playbook/base.py

```diff
@@ -589,10 +590,12 @@ def _validate_variable_keys(ds):
                 _validate_variable_keys(ds)
                 return combine_vars(self.vars, ds)
             elif isinstance(ds, list):
+                frame_info = inspect.stack()[1]
+                line_number = frame_info.lineno
```

Comment:
This will always show the same line of code, what you want to show is the line/file of the 'ds' object, which we keep and display internally already in other places

Reply:
To expand on what @bcoca mentions, we want to show the filename and line number of the YAML file containing the issue, and not the python file the warning came from.  Instead of using `inspect`, you will just want to use the `ds` variable.  The properties available on the `ds` object are defined at:

https://github.com/ansible/ansible/blob/e07b4edc547e2a5bd429d1027c0102235616db6c/lib/ansible/parsing/yaml/objects.py#L28-L53